installer_fabfile.py

```diff
@@ -293,16 +298,21 @@ def cloud_init(gs_tenant,
             if stack_password is None:
                 stack_password = prompt(
                     cyan("Enter Openstack user password: "))
+
             if stack_auth_url is None:
-                while stack_auth_url is None or \
-                        not stack_auth_url.endswith(CLOUD_AUTH_URL_VERSION):
-                    stack_auth_url = prompt(
-                        cyan("Enter OpenStack auth URL base "
-                             "(ex: http://10.10.10.10:5000/v2.0): "))
-                    if not stack_auth_url.endswith(CLOUD_AUTH_URL_VERSION):
-                        stack_auth_url = re.sub('/v2.0[/]?$|/v3[/]?$',
-                                                CLOUD_AUTH_URL_VERSION,
-                                                stack_auth_url)
+                stack_auth_url = \
+                    prompt(cyan("Enter OpenStack auth URL "
+                                "(eg: http://10.10.10.10:5000/): "))
+
+                if re.search(AUTH_URL_VERSION_LIKELY, stack_auth_url[-9:]):
+                    # The user shouldn't have included the version segment, but
+                    # did anyway. Remove it.
+                    version_index = re.search(AUTH_URL_VERSION_LIKELY,
+                                              stack_auth_url)
+                    stack_auth_url = stack_auth_url[:version_index.start()]
+
+                # Append our version number to the base URL.
+                stack_auth_url = os.path.join(stack_auth_url, AUTH_URL_VERSION)
```

Comment:
The intent of the changed code block was to allow the user to copy/paste the entire value of OS_AUTH_URL from their cloud generated keystonerc_admin file.  I believe that approach will be less error prone than requiring them to input the partial (base) part of the string.

Suggestion is to revert this change, but change the prompt string to "Enter OpenStack auth URL (eg: http://10.10.10.10:5000/v2.0):"